airflow/providers/amazon/aws/triggers/redshift_cluster.py

```diff
@@ -70,6 +76,18 @@ async def run(self) -> AsyncIterator["TriggerEvent"]:
                     else:
                         if self.attempts < 1:
                             yield TriggerEvent({"status": "error", "message": f"{self.task_id} failed"})
+                elif self.operation_type == "delete_cluster":
+                    response = await hook.delete_cluster(
```

Comment:
The parameter `attempts` is not being checked here, and is ignored inside `delete_cluster`, which runs until an error or the expected state is reached. One problem with this is that line 94 checks the `attempts` parameter before yielding a `TriggerEvent`. If an exception is thrown at any point inside this operation, it will be caught, then ignored because `self.attempts` will not have changed.

Reply:
thanks for pointing it out. added a check